### PR TITLE
Improve performance for column replicate for uint32 type. In our benc…

### PR DESCRIPTION
**Pull request by zzachimed**
Changelog category (leave one):

- Improvement

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
use simd to re-write the current column replicate funcion and got  2x performance boost in our unit benchmark test

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/
use simd to re-write the current column replicate funcion and got  2x performance boost in our unit benchmark test

### CODE REVIEW

**Comment by UnamedRus** [comment]
Related https://github.com/ClickHouse/ClickHouse/issues/30979

It's possible to have similar optimization for other (U)Int* types?

**Comment by zzachimed** [comment]
> Related #30979
> 
> It's possible to have similar optimization for other (U)Int* types?

step by step ,we will afford similar optimization for other (U)Int* types for following commits

**Comment by zzachimed** [comment]
hi CI maitainer :
 could you help me to check that  lable check fail issue that break my CI test . it always report "User 'zzachimed' is not trusted"

1s
Run cd "$GITHUB_WORKSPACE/tests/ci"
INFO:root:Getting URL with 5 and sleep 0 in between: https://api.github.com/repos/ClickHouse/ClickHouse/pulls/37235
INFO:root:Getting URL with 5 and sleep 0 in between: https://api.github.com/users/zzachimed/orgs
INFO:root:Getting URL with 5 and sleep 0 in between: https://github.com/ClickHouse/ClickHouse/pull/37235.diff
Fetched info about 2 changed files
Got labels {'pr-improvement'}
INFO:root:User 'zzachimed' is not trusted
change labels: add [], remove []
Notice: Cannot run
Error: Process completed with exit code [1](https://github.com/ClickHouse/ClickHouse/runs/6447746246?check_suite_focus=true#step:4:1).

**Comment by kitaisreal** [comment]
@Mergifyio update

File: src/Columns/ColumnVector.h [hunk 1]
```diff
@@ -136,7 +136,9 @@ class ColumnVector final : public COWHelper<ColumnVectorHelper, ColumnVector<T>>
 
     /// Sugar constructor.
     ColumnVector(std::initializer_list<T> il) : data{il} {}
-
+    #ifdef __SSE4_2__
+    ColumnPtr replicateSSE2(const IColumn::Offsets & offsets) const;
```
Comment (alexey-milovidov) [inline src/Columns/ColumnVector.h:140]: It is strange that the function is named 'SSE2' while it is using SSE 4.2 and looks like SSSE3 is enough.

Comment (zzachimed) [reply src/Columns/ColumnVector.h:140]: ok i will change it.

File: src/Columns/ColumnVector.cpp [hunk 2]
```diff
@@ -548,6 +553,107 @@ ColumnPtr ColumnVector<T>::replicate(const IColumn::Offsets & offsets) const
...
+ColumnPtr ColumnVector<T>::replicateSSE2(const IColumn::Offsets & offsets) const
+{
+    auto res = this->create(offsets.back());
+
+    auto it = res->getData().begin(); // NOLINT
+
+    /// Column is using PaddedPODArray, so we don't have to worry about the 4 out of range elements.
+
+    size_t prev_offset = 0;
+    int copy_begin = -1;
```
Comment (alexey-milovidov) [inline src/Columns/ColumnVector.cpp:569]: The offset can be larger than int max.

Comment (zzachimed) [reply src/Columns/ColumnVector.cpp]: ok I will fix this "offset " issue.

**Comment by zzachimed** [comment]
@kitaisreal :
hi CI maintainer, Could you help  me to check  case "[IntegrationTestsTsan2](https://github.com/ClickHouse/ClickHouse/runs/6545972703?check_suite_focus=true#logs)"  2022-05-22 20:13:06,977 Totally have 0 with status FAILED . It seems my code is ok. it caused by container oom .other case all pass . Could you help me pass it or re-run this single case.

**Comment by kitaisreal** [comment]
@zzachimed seems like CI issue, not related to changes. Can we somehow write SQL test to see performance improvement ?

File: src/Columns/ColumnVector.cpp [hunk 3]
```diff
@@ -548,6 +553,110 @@ ColumnPtr ColumnVector<T>::replicate(const IColumn::Offsets & offsets) const
     return res;
 }
 
+
+#ifdef __SSE2__
+
+template <typename T>
+ColumnPtr ColumnVector<T>::replicateSSE2(const IColumn::Offsets & offsets) const
```
Comment (kitaisreal) [inline src/Columns/ColumnVector.cpp:560]: @zzachimed please write high level description comment what we try to archive here.

**Review by kitaisreal — CHANGES_REQUESTED** [review]

**Review by kitaisreal — APPROVED** [review]

**Pull request merged**